### ISSUE TITLE
[package] openssl/1.1.1k: PR 4742 made TLS not work with self-signed ceretificates

### ISSUE THREAD

**Issue by Talkless** [conan-io/conan-center-index#5662]
I've bisected commit 22152096a24d11ad17c08f2a0fa36d38047af19c which somehow broke our use case  with OpenSSL, where we use Qt 5.13.2 (yes, stuck on that old) `QtNetwork` module with `QSslSocket`, where custom self-signed CA certificate is set via `QSslConfiguration`, and both server and client has (signed by that self-signed CA) their own client/server certificates.

Since newly built OpenSSL library from Conan, our server-client pair stopped working - sockets just timeouts, failing to handshake or whatever...

- stdout/stderr does not print anything interesting
- Not even with `QT_LOGGING_RULES` set to `*.debug=true`
- I don't see anything interesting in strace/ltrace, and nothing about `/etc/ssl` path.

The workaround I've found is to set:
```
[options]
openssl:openssldir=/nonexistant
```
Workaround works for `1.1.1i` we used originally before breaking upgrade, and with latest `1.1.1k.`

If I manually set `openssl:openssldir=/etc/ssl` (as per default set in that PR), `QSslSocket` also stops working.

This self-signed setup worked for us for quite some time, even before we started using OpenSSL from Conan...

### Package and Environment Details (include every applicable attribute)
  * Package Name/Version: **openssl/1.1.1i, openssl/1.1.1k**
  * Operating System+version: **Debian 10 (buster) amd64**
  * Compiler+version: **GCC 8.3.0**
  * Conan version: **conan 1.36.0**
  * Python version: **Python 3.7.3**

### Conan profile (output of `conan profile show default` or `conan profile show <profile>` if custom profile is in use)
```
[settings]
os=Linux
os_build=Linux
arch=x86_64
arch_build=x86_64
compiler=gcc
compiler.version=8
compiler.libcxx=libstdc++
build_type=Release
[options]
[build_requires]
[env]
```
```
include(default)

[settings]
compiler.libcxx=libstdc++11
```

**Comment by perseoGI** [comment]
Hi there!
I'm reviewing old issues and found yours.
`openssl/1.1.1k` is not supported anymore by Conan Center Index.
I'll close this issue and if for some reason, this problem continues in newer versions, do not hesitate to open a new ticket mentioning me.

Happy coding 🐸

**Comment by perseoGI** [comment]
Hi there, 
I was reviewing old issues and I have come across this one.
The `openssl/1.1.1k` is no longer supported by Conan Center Index.

I'm closing this issue as this recipe has suffer plenty of modifications since this issue was reported.
If for some reason the problem remains nowadays, do not hesitate to reopen this issue or create a new ticket with updated information, preferably using `conan v2`.

Happy coding 🐸
